### PR TITLE
Clean up StyleImpl

### LINKED ISSUE #392: Reduce duplication in StyleImpl
[`StyleImpl`](https://github.com/KyoriPowered/adventure/blob/master/api/src/main/java/net/kyori/adventure/text/format/StyleImpl.java) has a lot of duplication when it comes to `TextDecoroation` - it would be nice to find a way to reduce this.

### PR DESCRIPTION
**Pull request by MrMacor**
This closes #392 by replacing the hard-coded `TextDecoration` values in `StyleImpl` and `StyleImpl.BuilderImpl` with `TextDecoration.State[]`s. The index of each `State` is the same as the ordinal of the `TextDecoration` it represents.

### CODE REVIEW

File: api/src/main/java/net/kyori/adventure/text/format/Style.java [hunk 1]
```diff
@@ -108,7 +109,7 @@
    */
   static @NotNull Style style(final @Nullable TextColor color) {
     if (color == null) return empty();
-    return new StyleImpl(null, color, TextDecoration.State.NOT_SET, TextDecoration.State.NOT_SET, TextDecoration.State.NOT_SET, TextDecoration.State.NOT_SET, TextDecoration.State.NOT_SET, null, null, null);
+    return new StyleImpl(null, color, Stream.of(TextDecoration.values()).map(decoration -> TextDecoration.State.NOT_SET).toArray(TextDecoration.State[]::new), null, null, null);
```
Comment (zml2008) [inline api/src/main/java/net/kyori/adventure/text/format/Style.java:112]: this is a joke, right?

**Review by zml2008 — CHANGES_REQUESTED** [review]

**Comment by kashike** [comment]
I am not really a fan of this approach - it introduces mutable state (via modifying the arrays) an immutable object.

**Comment by zml2008** [comment]
I don't think this will be something we'll be wanting to merge.

**Pull request closed without merging**